std/assembly/typedarray.ts

```diff
@@ -1734,27 +1734,26 @@ function REVERSE<TArray extends ArrayBufferView, T>(array: TArray): TArray {
 // @ts-ignore: decorator
 @inline
 function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset: i32 = 0, length: i32 = -1): TArray {
+  var byteLength: i32;
   var bufferByteLength = buffer.byteLength;
-  if (<u32>byteOffset >= <u32>bufferByteLength) {
+  // @ts-ignore: cast
+  if ((<u32>byteOffset > <u32>bufferByteLength) | (byteOffset & (sizeof<T>() - 1))) {
```

Comment:
Computing from alignment seems more natural here. Perhaps move the original `const mask = <i32>(1 << alignof<T>() - 1);` to the top of the function, and reuse it here and when checking bufferByteLength below?

Reply:
Also, wrapping the first part in `i32(...)` might avoid the need for a `@ts-ignore`

Reply:
1) `<i32>(1 << alignof<T>() - 1)` had a problem with operation priority. It should be `(1 << alignof<T>()) - 1`
2) `1 << alignof<T>()` looks redundant and better use `sizeof<T>() - 1` which I think compiled little bit faster due to not need const folding

Reply:
Mostly proposing a `const` because it precomputes, while an inline `whatev<T>() +-*/ 1` is not guaranteed. We precompute the latter currently, but we do this trying to eliminate the branch in the first place, so a `const` is a safer bet.

Reply:
ok. done!